qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -0,0 +1,203 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The HFInitialPoint class to compute an initial point for the VQE Ansatz parameters."""
+
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+
+from qiskit_nature.exceptions import QiskitNatureError
+from qiskit_nature.properties.second_quantization.electronic import ElectronicEnergy
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.properties.second_quantization.second_quantized_property import (
+    GroupedSecondQuantizedProperty,
+)
+
+from .initial_point import InitialPoint
+
+logger = logging.getLogger(__name__)
+
+
+class HFInitialPoint(InitialPoint):
+    """Compute the Hartree-Fock (HF) initial point.
+
+    All excitations will have zero coefficient and energy correction as the all-zero initial point
+    corresponds to the HF initial point once the HF initial state is prepended to the circuit.
```

Comment:
I think this phrasing needs to clarify that this only holds for the UCC ansatz. I don't think that in general this applies to any ansatz.